AppBuilder/platform/ABDataCollection.js

```diff
@@ -585,6 +585,30 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
       });
    }
 
+   warningsAll() {
+      // report both OUR warnings, and any warnings from any of our fields
+      const allWarnings = [].concat(this._warnings);
+
+      if (!this.datasource) {
+         allWarnings.push({ message: "I got no the datasourse.", data: {} });
```

Comment:
lets make this descriptive like "Datacollection[${this.label}][${this.id}] doesn't have a datasource."